README.md

```diff
@@ -1,4 +1,208 @@
+
+    override fun CoroutineScope.start(
+        input: ReceiveChannel<CalculatorAction>,
+        stateAccessor: StateAccessor<Int>,
+        output: SendChannel<CalculatorAction>,
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() == 9) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Now we need to modify how store is created by adding this `sideEffect`:
+```kotlin
+val storeWithSideEffect = GlobalScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions - 
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Add(9)) 
+// Current state will become 9
+storeWithSideEffect.dispatch(CalculatorAction.Add(1)) 
+// Current state will become 10 and network request will happen
+```
+
+Library provides useful simplified versions of `SideEffect` interface:
+- `SimpleSideEffect` that produces on one input action either one output action or no action.
+- `CancellableSideEffect` that cancels currently executed `Job` and starts a new one.
+
+### Implementation details
+
+By default, CoRedux starts main reducer coroutine ("manager"), when `StateReceiver` subscribes to `CoreduxStore` instance.
+Reducer coroutine itself starts all side effects coroutines ("workers"), sets current state to initial state
+and immediately updates with it `StateReceiver`. Then reducer coroutine starts listening
+for a new action from actions channel. 
+
+When a new action received, reducer coroutine triggers reducer function, that updates current state inside coroutine,
```

Comment:
> new action  **is**  received.

> that updates current state inside coroutine
I don't understand that? It changes state of coroutine or do you mean the state of the whole store?

Reply:
> I don't understand that? It changes state of coroutine or do you mean the state of the whole store?

It changes the current state field inside reducer coroutine.

Reply:
For me as reader it doesn't add value, it rather confuses me I guess ... 

Up to you  ...

Reply:
I've reworked "Implementation details" section.